### PR TITLE
update django admin list view for enterprise customer model

### PR DESCRIPTION
**Pull request by zubair-arbi**
**Description:** update django admin list view for enterprise customer model to:
1. ~~Display Enterprise Customer name only in green message bar at top of screen (versus "`<Enterprise Customer uuid-asf123das-d1fsad-fa32se: name>`"~~
2. Replace "logo" column values in list table with a checkbox
3. Add "consent" column with checkboxes for values
4. Replace identity provider column values with checkboxes
5. Replace catalog column values with checkboxes
6. Replace coupon column values with checkboxes

**JIRA:** [ENT-396](https://openedx.atlassian.net/browse/ENT-396)

**Dependencies:** N/A

**Merge deadline:** N/A

**Installation instructions:** Install `edx-enterprise` from this branch `zub/ENT-396-update-enterprise-customer-admin-list-view` in `edx-platform`.

**Testing instructions:**

1. Go to django admin list view page for "Enterprise Customers" and view the checkboxes (boolean values) against the columns `Logo`, `Enable data sharing consent`, `Identity provider`, `Enterprise catalog` and `Ecommerce coupons`.

screen shot 2018-01-05 at 5 46 11 pm

**Merge checklist:**

- [ ] Check that the versions of the requirements in the `platform-master.in` file match edx-platform.
- [ ] New requirements are in the right place (`base.in` if only used in enterprise; in the correct `platform-****.in` files if they're hosted in edx-platform)
- [ ] Regenerate requirements with `make upgrade && make requirements` (and make sure to fix any errors).
  **DO NOT** just add dependencies to `requirements/*.txt` files.
- [ ] Called `make static` for webpack bundling if any static content was updated.
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are (reasonably) squashed
- [ ] Translations are updated
- [ ] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is finished.
- [ ] Delete working branch (if not needed anymore)
- [ ] edx-platform PR (be sure to include edx-platform requirements upgrades that were present in this PR)

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

### CODE REVIEW

**Comment by mattdrayer** [comment]
I think we can rename the "Enable data sharing consent" column to "Enable DSC"

**Comment by zubair-arbi** [comment]
@mattdrayer I have renamed the "Enable data sharing consent" column to "Enable DSC".
As for the remaining AC

> 1. Display Enterprise Customer name only in green message bar at top of screen (versus "`<Enterprise Customer uuid-asf123das-d1fsad-fa32se: name>`"

Django uses the string representation of the model in the messaging on admin page and we have implemented our custom `__str__` method on the `EnterpriseCustomer` model to make the string representation of its instances more detailed (include both `uuid` and `name`).

If its required I can update the `__str__` method to only contain name. Let me know your thought on this.

**Comment by mattdrayer** [comment]
Okay, if this is a native Django thing there is no need to customize -- we can keep the behavior as-is.

File: enterprise/admin/__init__.py [hunk 1]
```diff
@@ -111,61 +111,57 @@ class EnterpriseCustomerAdmin(DjangoObjectActions, SimpleHistoryAdmin):
...
+    def has_ecommerce_coupons(self, instance):
         """
-        Instance is EnterpriseCustomer. Return e-commerce coupon urls.
+        Return True if provded enterprise customer has ecommerce coupons.
+
+        Arguments:
+            instance (enterprise.models.EnterpriseCustomer): `EnterpriseCustomer` model instance
         """
-        coupon_urls = ''
         entitlements = instance.enterprise_customer_entitlements.all()
```
Comment (saleem-latif) [inline enterprise/admin/__init__.py]: you can simply do this
```
return instance.enterprise_customer_entitlements.exists()
```

File: enterprise/admin/__init__.py [hunk 2]
```diff
@@ -176,49 +172,66 @@ def get_form(self, request, obj=None, **kwargs):
...
-        Enterprise catalog id with a link to catalog details page.
+        Return True if EnterpriseCustomer has related identity provider.
 
         Arguments:
             instance (enterprise.models.EnterpriseCustomer): `EnterpriseCustomer` model instance
+        """
+        identity_provider = instance.enterprise_customer_identity_provider
+
+        has_identity_provider = False
+        if identity_provider and (identity_provider.provider_id or identity_provider.provider_name):
```
Comment (saleem-latif) [inline enterprise/admin/__init__.py]: Isn't `provider_id` a required field of `EnterpriseCustomerIdentityProvider` ? if so I think we can simply do this.
```
return hasattr(instance.enterprise_customer_identity_provider)
```

File: enterprise/admin/__init__.py [hunk 3]
```diff
@@ -176,49 +172,66 @@ def get_form(self, request, obj=None, **kwargs):
...
+
+    def has_enterprise_catalog(self, instance):
+        """
+        Return True if EnterpriseCustomer has catalog id with a link to catalog details page.
+
+        Arguments:
+            instance (enterprise.models.EnterpriseCustomer): `EnterpriseCustomer` model instance
         """
-        # Return None if EnterpriseCustomer does not have an associated catalog.
         if not instance.catalog:
```
Comment (saleem-latif) [inline enterprise/admin/__init__.py]: nit
```
return instance.catalog is not None
```

**Comment by zubair-arbi** [comment]
@saleem-latif I have addressed your feedback.
@mattdrayer have another look on the changes.

**Pull request merged**